### PR TITLE
Check if test executions have debug info

### PR DESCRIPTION
**Pull request by petertrr**
* Add flag into TestExecutionDto
* Set flag in some endpoints

Part of #515 only on backend

### CODE REVIEW

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt [hunk 1]
```diff
@@ -49,24 +63,43 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
...
+        example = "basic"
+    )
     @Suppress("LongParameterList", "TOO_MANY_PARAMETERS", "TYPE_ALIAS")
     fun getTestExecutions(
         @RequestParam executionId: Long,
         @RequestParam page: Int,
         @RequestParam size: Int,
         @RequestParam(required = false) status: TestResultStatus?,
         @RequestParam(required = false) testSuite: String?,
+        @RequestParam(required = false, defaultValue = "false") checkDebugInfo: Boolean,
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:83]: Why defaultValue = "false"?

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt (hunk 1 above)
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:83]: We should perform additional operations (here - some filesystem queries) only if it's explicitly requested by the consumer

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt [hunk 2]
```diff
@@ -49,24 +63,43 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
      * @param status
      * @param testSuite
      * @param authentication
+     * @param checkDebugInfo if true, response will contain information about whether debug info data is available for this test execution
      * @return a list of [TestExecutionDto]s
      */
     @GetMapping(path = ["/api/$v1/testExecutions"])
+    @Parameter(
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:70]: Good idea, added

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt (hunk 1 above)
Comment (kgevorkyan) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:72]: But if it's not requested, but debug info doesn't exist. we will get previous problem again?

Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:72]: It should be handled on frontend properly: if there is a possibility that debug info is requested, then its presence should be checked. If component doesn't show any related info, then it shouldn't care whether it's available or not.
Also - I've updated the PR and made this field nullable. This way we can distinguish cases when debug info is unavailable and when it hassn't been checked

Comment (kgevorkyan) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:72]: Didn't look, that this method used not only by frontend
Sure, of course, should be `false`

Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt:72]: It's used currently only by frontend, but in several places, only some of them are related to debug info. If this method is used by API integrations, then they also may need this information only someties

**Pull request merged**